scapy/contrib/gmlan.py

```diff
@@ -0,0 +1,704 @@
+                         lambda pkt: pkt.subfunction == 0x02),
+        ConditionalField(XByteField('DTCFailureType', 0),
+                         lambda pkt: pkt.subfunction == 0x02),
+
+    ]
+
+
+bind_layers(GMLAN, GMLAN_RFRD, service=0x12)
+
+
+class GMLAN_RFRDPR(Packet):
+    name = 'ReadFailureRecordDataPositiveResponse'
+    fields_desc = [
+        ByteEnumField('subfunction', 0, GMLAN_RFRD.subfunctions)
+    ]
+
+
+bind_layers(GMLAN, GMLAN_RFRDPR, service=0x52)
+
+
+class GMLAN_RFRDPR_RFRI(Packet):
+    __slots__ = Packet.__slots__ + ["_pre_dissect_len"]
+
+    def pre_dissect(self, s):
+        self._pre_dissect_len = len(s)
+        return s
+
+    failureRecordDataStructureIdentifiers = {
+        0x00: "PID",
+        0x01: "DPID"
+    }
+    name = 'ReadFailureRecordDataPositiveResponse_readFailureRecordIdentifiers'
+    fields_desc = [
+        ByteEnumField('failureRecordDataStructureIdentifier', 0,
+                      failureRecordDataStructureIdentifiers),
+        ConditionalField(XByteField('failureRecordNumber1', 0),
+                         lambda pkt: pkt._pre_dissect_len > 3),
+        ConditionalField(XByteField('DTCHighByte1', 0),
+                         lambda pkt: pkt._pre_dissect_len > 3),
+        ConditionalField(XByteField('DTCLowByte1', 0),
+                         lambda pkt: pkt._pre_dissect_len > 3),
+        ConditionalField(XByteField('DTCFailureType1', 0),
+                         lambda pkt: pkt._pre_dissect_len > 3),
+        ConditionalField(XByteField('failureRecordNumber2', 0),
+                         lambda pkt: pkt._pre_dissect_len > 7),
+        ConditionalField(XByteField('DTCHighByte2', 0),
+                         lambda pkt: pkt._pre_dissect_len > 7),
+        ConditionalField(XByteField('DTCLowByte2', 0),
+                         lambda pkt: pkt._pre_dissect_len > 7),
+        ConditionalField(XByteField('DTCFailureType2', 0),
+                         lambda pkt: pkt._pre_dissect_len > 7),
+        ConditionalField(XByteField('failureRecordNumber3', 0),
+                         lambda pkt: pkt._pre_dissect_len > 11),
+        ConditionalField(XByteField('DTCHighByte3', 0),
+                         lambda pkt: pkt._pre_dissect_len > 11),
+        ConditionalField(XByteField('DTCLowByte3', 0),
+                         lambda pkt: pkt._pre_dissect_len > 11),
+        ConditionalField(XByteField('DTCFailureType3', 0),
+                         lambda pkt: pkt._pre_dissect_len > 11),
+        ConditionalField(XByteField('failureRecordNumber4', 0),
+                         lambda pkt: pkt._pre_dissect_len > 15),
+        ConditionalField(XByteField('DTCHighByte4', 0),
+                         lambda pkt: pkt._pre_dissect_len > 15),
+        ConditionalField(XByteField('DTCLowByte4', 0),
+                         lambda pkt: pkt._pre_dissect_len > 15),
+        ConditionalField(XByteField('DTCFailureType4', 0),
+                         lambda pkt: pkt._pre_dissect_len > 15)
```

Comment:
Could you use a `PacketListField`, something like
```python
class GMLAN_DTC(Packet):
    name = "Fancy name"
    fields_desc = [XByteField('failureRecordNumber', 0),
                          XByteField('DTCHighByte', 0),
                          XByteField('DTCLowByte', 0),
                          XByteField('DTCFailureType', 0) ]
```
then
```python
PacketListField("name", [], GMLAN_DTC)
```

Reply:
Thank you very much. This is exactly what I was looking for.